classes/Product.php

```diff
@@ -5516,7 +5516,13 @@ public static function getProductProperties($id_lang, $row, Context $context = n
         $row['category'] = Category::getLinkRewrite((int) $row['id_category_default'], (int) $id_lang);
         $row['category_name'] = Db::getInstance()->getValue('SELECT name FROM ' . _DB_PREFIX_ . 'category_lang WHERE id_shop = ' . (int) $context->shop->id . ' AND id_lang = ' . (int) $id_lang . ' AND id_category = ' . (int) $row['id_category_default']);
         $row['link'] = $context->link->getProductLink((int) $row['id_product'], $row['link_rewrite'], $row['category'], $row['ean13']);
-        $row['manufacturer_name'] = !empty($row['id_manufacturer']) ? Manufacturer::getNameById((int) $row['id_manufacturer']) : null;
+        if (empty($row['manufacturer_name'])) {
+            if (isset($row['id_manufacturer']) && !empty((int) $row['id_manufacturer'])) {
+                $row['manufacturer_name'] = Manufacturer::getNameById((int) $row['id_manufacturer']);
+            } else {
+                $row['manufacturer_name'] = null;
+            }
+        }
```

Comment:
```suggestion
        if (empty($row['manufacturer_name']) && !empty((int) $row['id_manufacturer'])) {
            $row['manufacturer_name'] = Manufacturer::getNameById((int) $row['id_manufacturer']);
        } else {
            $row['manufacturer_name'] = null;
        }
```

Reply:
that should do it, right?

Reply:
it also works, in addition to gaining readability. I modify my copy